src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/IWarningLogger.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+#nullable disable
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+/// <summary>
+/// DO NOT use this to extend TestPlatform, it is public only because ITestPlatform is also public, and will be made internal later.
+/// </summary>
+public interface IWarningLogger
```

Comment:
We don't have a better way at the moment but maybe adding `Obsolete` would help.

Reply:
All the other apis would then become obsolete and that would be a nightmare. It is okay for translation layer where we don't implement it ourselves, but sadly not here.